src/test/ui/const-generics/issues/issue70273-assoc-fn.rs

```diff
@@ -0,0 +1,17 @@
+// check-pass
+
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+trait T<const A: usize> {
+    fn f();
+}
+struct S;
+
+impl T<0usize> for S {
+    fn f() {}
+}
+
+fn main() {
+    let _err = <S as T<0usize>>::f();
```

Comment:
`<S as T<123>>::f::<456>()` should also work, if you added a const param to `f` as well.

Reply:
will add a test

Reply:
I mean you'll need to implement it, it won't work with the current approach.

Reply:
for whatever reason adding a const param to f actually prevents the ice. not quite sure why, but this is new behavior so lets add a test for it
```rust
#![feature(const_generics)]
//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash

trait T<const A: usize> {
    fn l<const N: usize>() -> usize;
    fn r<const N: usize>() -> usize;
}

struct S;

impl<const N: usize> T<N> for S {
    fn l<const M: usize>() -> usize { N }
    fn r<const M: usize>() -> usize { M }
}

fn main() {
   assert_eq!(<S as T<123>>::l::<456>(), 123);
   assert_eq!(<S as T<123>>::r::<456>(), 456);
}
```

Reply:
Oh, oops, it's because both the trait and the method have the exact same number of parameters and the same types. Make one of them a `bool` instead of `usize`, then you should see how it thinks the method segment has the trait's parameter (which is wrong).

Reply:
jup. Quite surprisingly, using usize and u16 works :thinking: 
but using `bool` and `usize` does not work on nightly rn

Reply:
Yes, because the integer literals will work with any integer type.